flow/src/main/java/flow/Flow.java

```diff
@@ -146,8 +146,8 @@ public void goTo(final Path path) {
   }
 
   /**
-   * Reset to the specified screen. Pops until the screen is found. If the screen is not found,
-   * the entire backstack is replaced with the screen.
+   * Set the specified screen to the top of the backstack. If the screen already
+   * exists in the backstack, the screen will be moved to the top of the backstack.
```

Comment:
We really do pop back to an existing screen. This makes it sound like we remove it from the middle of the stack and move it to the top. 

The inaccuracy in the old comment was the second sentence: we don't replace the existing statck with the new screen, we just push the screen.